tool/src/cpp/formatter.rs

```diff
@@ -205,6 +205,92 @@ impl<'tcx> Cpp2Formatter<'tcx> {
     pub fn fmt_identifier<'a>(&self, name: Cow<'a, str>) -> Cow<'a, str> {
         self.c.fmt_identifier(name)
     }
+
+    pub fn fmt_c_api_callback_ret<'a>(
+        &self,
+        namespace: Option<String>,
+        method_name: String,
+        cpp_name: &'a str,
+    ) -> Cow<'a, str> {
+        let ns = namespace.unwrap_or("diplomat".to_string());
+        format!("{ns}::{CAPI_NAMESPACE}::DiplomatCallback_{method_name}_{cpp_name}_result").into()
+    }
+
+    pub fn fmt_callback_success_type(&self, success: &hir::SuccessType<hir::InputOnly>) -> String {
+        match success {
+            hir::SuccessType::Unit => "std::monostate".into(),
+            hir::SuccessType::OutType(o) => self.fmt_callback_out_type(o),
+            _ => unreachable!("unknown AST/HIR variant"),
+        }
+    }
+
+    pub fn fmt_callback_out_type(&self, ty: &hir::Type<hir::InputOnly>) -> String {
```

Comment:
issue: this is incorrect; it should be a `gen_` method since it may need to generate imports.

Is there a reason `gen_type_name()` does not work for this? We should not need to add any such large matches to support Result.